electrolytes/solvation_shell_utils.py

```diff
@@ -0,0 +1,152 @@
+import re
+import numpy as np
+import random
+from typing import List, Set, Callable
+
+from schrodinger.structutils import analyze
+from schrodinger.application.matsci import clusterstruct
+from schrodinger.comparison import are_conformers
+from schrodinger.application.jaguar.utils import group_with_comparison
+from schrodinger.comparison.atom_mapper import ConnectivityAtomMapper
+from schrodinger.structutils import rmsd
+from schrodinger.structure import Structure
+
+
+def filter_shells_with_solute_atoms(
+    shells: List[Set[int]], st: Structure, solute_res_names: List[str]
+) -> List[Structure]:
+    """
+    Filter out shells that contain solute atoms.
+    Args:
+        shells: List of sets of atom indices that correspond to shells
+        st: Entire structure from the PDB file
+        solute_res_names: List of residue names that correspond to solute atoms in the simulation
+    Returns:
+        List of Schrodinger structures that do not contain solute atoms
+    """
+    actual_shells = [st.extract(shell, copy_props=True) for shell in shells]
+    filtered_shells = [
+        shell
+        for shell in actual_shells
+        if not any(
+            at.getResidue().pdbres.strip() in solute_res_names for at in shell.atom
+        )
+    ]
```

Comment:
```suggestion
    solute_atoms = analyze.evaluate_asl(st, f'res {",".join(solute_res_names)}')
    filtered_shells = [shell for shell in shells if not shell.intersection(solute_atoms)]
```

evaluate_asl on the names of the solute residues gets all atom indices that are in residues with that name.

Reply:
changed